addons/mail/tests/test_mail_template.py

```diff
@@ -170,6 +170,16 @@ def test_server_archived_usage_protection(self):
         server.action_archive()  # No more usage -> can be archived
         self.assertFalse(server.active)
 
+    def test_mail_template_parse_partner_to(self):
+        partner = self.mail_template._parse_partner_to('1')
+        self.assertListEqual(partner, [1])
+
+        partners = self.mail_template._parse_partner_to('1,2,3')
+        self.assertListEqual(partners, [1, 2, 3])
+
+        partners = self.mail_template._parse_partner_to('res.partner(1, 2, 3)')
+        self.assertListEqual(partners, [2])
```

Comment:
Could be done using subTest and loops, easing addition of tests, like
```suggestion
        for partner_to, expected in [
            ('1', [1]),
            ('1,2,3', [1, 2, 3]),
            ('1, 2,  3', [1, 2, 3]),  # remove spaces
            ('[1, 2, 3]', [1, 2, 3]),  # %r of a list
            ('(1, 2, 3)', [1, 2, 3]),  # %r of a tuple
            ('(1, "wrong", 2, "partner_name", 3)', [1, 2, 3]),  # fault tolerant
            ('res.partner(1, 2, 3)', [2]),  # invalid input but avoid crash
        ]:
            with self.subTest(partner_to=partner_to):
                parsed = self.mail_template._parse_partner_to('1')
                self.assertListEqual(parsed, expected)
```